### PR TITLE
Add SoftKomik

### PR DESCRIPTION
**Pull request by MikeZeDev**

### CODE REVIEW

File: web/src/engine/websites/SoftKomik.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+            data: APIManga[]
+        }
+    }
+}
+
+@Common.ImageAjax(true)
+export default class extends DecoratableMangaScraper {
+
+    private nextBuild = '';
+    private readonly CDN = 'https://soft1.softdevices.my.id'; // other CDN is 'https://soft2.b-cdn.net' make an option?
```
Comment (ronny1982) [inline web/src/engine/websites/SoftKomik.ts:45]: Consider using array of CDNs

**Review by ronny1982 — APPROVED** [review]

File: web/src/engine/websites/SoftKomik.ts (hunk 1 above)
Comment (MikeZeDev) [reply web/src/engine/websites/SoftKomik.ts]: changed it to try both

File: web/src/engine/websites/SoftKomik.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        const request = new FetchRequest(url);
+        const { pageProps: { data: { imgSrc } } } = await FetchJSON<APIPages>(request);
+        return imgSrc.map(page => new Page(this, chapter, new URL(page, this.CDN[0]), { alternativeUrl: new URL(page, this.CDN[1]).href}));
+    }
+
+    public override async FetchImage(page: Page, priority: Priority, signal: AbortSignal): Promise<Blob> {
+        let blob = await Common.FetchImageAjax.call(this, page, priority, signal, true);
+        if (!blob.type.startsWith('image')) {
+            const fakepage = new Page(this, page.Parent as Chapter, new URL(page.Parameters.alternativeUrl as string));
+            blob = await Common.FetchImageAjax.call(this, fakepage, priority, signal, true);
```
Comment (ronny1982) [inline web/src/engine/websites/SoftKomik.ts:104]: May return promise directly instead of await and assign

Comment (MikeZeDev) [reply web/src/engine/websites/SoftKomik.ts:104]: ok

**Pull request merged**